ms_office_macro_bleach/bleach.py

```diff
@@ -0,0 +1,76 @@
+"""This module removes any and all macros/dynamic content from MS Office files.
+
+It is part of a suite of programs developed by the AntiMalware Alliance.
+
+Visit https://github.com/Anti-Malware-Alliance for more details about our organisation and projects.
+
+Feel free to contact benjamin.mcgregor2002@gmail.com for any questions regarding this module.
+"""
+from argparse import ArgumentParser
+from os import rename, path, remove
+from zipfile import ZipFile
+from shutil import make_archive, rmtree
+
+
+# Unzip file function
+# Converts file to a .zip archive and extracts it to file directory
+def unzip_file(file):
+
+    # Convert to zip archive
+    rename(file, file + ".zip")
+
+    # Extract contents into temporary folder
+    with ZipFile(file + ".zip", 'r') as zip_ref:
+        zip_ref.extractall(file + "_temp")
+
+    # Delete original file
+    remove(file + ".zip")
+
+
+# Remove macros function
+# Checks for macro files and deletes them if found
+def remove_macros(file, notify):
+    macros_found = False
+
+    # Deletes macro binary file if found
+    if path.exists(file + "_temp/word/vbaProject.bin"):
+        remove(file + "_temp/word/vbaProject.bin")
+        macros_found = True
+
+    # Deletes macro XML file if found
+    if path.exists(file + "_temp/word/vbaData.xml"):
+        remove(file + "_temp/word/vbaData.xml")
+        macros_found = True
+
+    # Notifies user of macro status if '-c' flag is present
+    if notify:
+        if macros_found:
+            print("Macros detected and removed.")
+        else:
+            print("No macros detected.")
+
+
+# Rezip function
+# Re-zips the unzipped file and restores its original file extension
+def rezip_file(file):
+
+    # Zip bleached folder
+    make_archive(file, "zip", file + "_temp")
+
+    # Convert back into original file format
+    rename(file + ".zip", file)
+
+    # Delete temporary folder
+    rmtree(file + "_temp")
+
+
+# Argument parser
+parser = ArgumentParser()
+parser.add_argument("file", help="file to be bleached")
+parser.add_argument("-c", help="notify if macros or potentially dangerous content is found", action="store_true")
+args = parser.parse_args()
+
+# Bleaching
+unzip_file(args.file)
```

Comment:
Before Bleaching the file, you like want to call a function to validate if the file is one of the supported extensions, otherwise skip bleaching.